### PR TITLE
added functionality for getting last watched episode id

### PR DESCRIPTION
**Pull request by MarekTalpas**
I made a function in local-storage.js for getting last watched episode id. and created static function, which returns either last visited episode id or first episode id within one series.
But problem is that once you save some episode into local storage; episodes (ids) take one place and other local storage saves only replace episode (id) on same (previos) place -> upodate them.

### CODE REVIEW

File: src/components/video-player-container/parts/video-player.js [hunk 1]
```diff
@@ -50,8 +50,11 @@ class VideoPlayer extends React.Component {
     return `${minutes}:${seconds}`;
   }
   static saveVideoTime(player, id) {
-    const timeProgress = Math.round(player.currentTime());
-    saveVideoProgress(id, timeProgress);
+    const currentTime = Math.round(player.currentTime());
+    const duration = Math.round(player.duration());
+    const diff = duration - currentTime;
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.js]: more like
```
const percentage = Math.round(currentTime / Math.round(duration / 100));
saveVideoProgress(id, percentage);
```

File: src/api/local-storage.js [hunk 2]
```diff
@@ -33,3 +34,17 @@ export function getProgressTimeById(id: number) {
   }
   return 0;
 }
+
+export function getLastEpisodeID(eIDs: Array<number>) {
+  const data: ?string = localStorage.getItem(HISTORY_LIST);
+  const parsedData = JSON.parse(data || 'null');
+  if (parsedData) {
+    const lastEpisode =
+      parsedData.reverse().find(histEntry => eIDs.find(id => histEntry.episodeId === id));
```
Comment (varholak-peter) [inline src/api/local-storage.js]: Why are we reversing here? I think we agreed we won't be counting on the ordering since it doesn't represent actual time of viewing.

File: src/api/local-storage.js [hunk 3]
```diff
@@ -33,3 +34,17 @@ export function getProgressTimeById(id: number) {
   }
   return 0;
 }
+
+export function getLastEpisodeID(eIDs: Array<number>) {
```
Comment (varholak-peter) [inline src/api/local-storage.js]: I would prefer a better name for this function. What last episode?

File: src/api/local-storage.js [hunk 4]
```diff
@@ -33,3 +34,17 @@ export function getProgressTimeById(id: number) {
...
+  const data: ?string = localStorage.getItem(HISTORY_LIST);
+  const parsedData = JSON.parse(data || 'null');
+  if (parsedData) {
+    const lastEpisode =
+      parsedData.reverse().find(histEntry => eIDs.find(id => histEntry.episodeId === id));
+    if (lastEpisode) {
+      return (lastEpisode.progressTime <= 95) ? lastEpisode.episodeId : `finished-${lastEpisode.episodeId}`;
```
Comment (varholak-peter) [inline src/api/local-storage.js]: Two things:
1) no magic numbers
2) This whole thing should be part of the find function. If the user watched the first episode it would just return 'finished-X' which is totally not what we want.

File: src/components/series-detail/series-container.js [hunk 5]
```diff
@@ -86,7 +105,8 @@ class SeriesContainer extends React.Component {
   }
   state: SeriesContainerState
   componentWillMount() {
-    getSeriesByID(this.props.params.id)
+    const { id: serieID } = this.props.params;
+    getSeriesByID(serieID)
```
Comment (varholak-peter) [inline src/components/series-detail/series-container.js]: this is an unnecessary change

File: src/api/local-storage.js [hunk 6]
```diff
@@ -33,3 +36,31 @@ export function getProgressTimeById(id: number) {
   }
   return 0;
 }
+
+export function getLastWatchedEpisodeID(episodeIDs: Array<number>) {
```
Comment (garySiladi) [inline src/api/local-storage.js]: I'm not sure if your function does the job or not, but I'm sure it's way too bloated for its objective. Check this solution out:
```
export function getLastWatchedEpisodeID(episodeIDs: Array<number>) {
  const data: ?string = localStorage.getItem(HISTORY_LIST);
  const historyData = JSON.parse(data || 'null');
  if (historyData) {
    const result = episodeIDs.find(episodeID =>
      history.every(
        histItem => histItem.id !== episodeID || (histItem.id === episodeID && histItem.progress < 95)
      )
    );
  }
  return result || episodeIDs[0];
}
```
It suites our needs well. You can test its functionality here: https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&targets=&browsers=&builtIns=false&debug=false&code=const%20history%20%3D%20%5B%0A%20%20%7B%0A%20%20%20%20id%3A%2010%2C%0A%20%20%20%20progress%3A%2010%2C%0A%20%20%7D%2C%0A%20%20%7B%0A%20%20%20%20id%3A%2036%2C%0A%20%20%20%20progress%3A%2097%2C%0A%20%20%7D%2C%0A%20%20%7B%0A%20%20%20%20id%3A%20154%2C%0A%20%20%20%20progress%3A%2090%2C%0A%20%20%7D%2C%0A%5D%3B%0A%0Aconst%20episodeIDs%20%3D%20%5B36%2C%20154%2C%2010%5D%3B%0A%0Aconst%20result%20%3D%20episodeIDs.find(episodeID%20%3D%3E%20%0A%20%20history.every(%0A%20%20%20%20histItem%20%3D%3E%20histItem.id%20!%3D%3D%20episodeID%20%7C%7C%20(histItem.id%20%3D%3D%3D%20episodeID%20%26%26%20histItem.progress%20%3C%2095))%0A)%3B%0A%0Aconsole.log(result%20%7C%7C%20episodeIDs%5B0%5D)%3B

File: package.json [hunk 7]
```diff
@@ -57,7 +57,11 @@
     "whatwg-fetch": "2.0.2"
   },
   "dependencies": {
+    "babel-cli": "^6.24.1",
```
Comment (garySiladi) [inline package.json]: remove all these added packages

File: src/components/video-player-container/parts/video-player.js (hunk 1 above)
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.js]: ^^^ what about this?

File: src/components/video-player-container/parts/video-player.js [hunk 8]
```diff
@@ -5,7 +5,7 @@ import 'video.js/dist/video-js.css';
 import videojs from 'video.js';
 import VideoPlayerInterface from './video-player-controls';
 import './video-player.css';
-import { saveVideoProgress, getProgressTimeById } from '../../../api/local-storage';
+import * as LS from '../../../api/local-storage';
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.js]: I would import them as we had until now, there are not so many for the wildcard use.

File: src/components/video-player-container/parts/video-player.js [hunk 9]
```diff
@@ -50,14 +50,25 @@ class VideoPlayer extends React.Component {
...
+    const diff = duration - currentTime;
+    const currentTimePerc = 100 - Math.round((diff / duration) * 100);
+    LS.saveVideoProgress(id, currentTimePerc);
   }
   static createVideoSaver(player, videoId) {
     return setInterval(() => {
       VideoPlayer.saveVideoTime(player, videoId);
     }, 1000);
   }
+  static getEpisodesIDs(allSeries) {
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.js]: don't add this function into this file ... just in the `series-container`

**Comment by garySiladi** [comment]
add also tests

File: src/components/video-player-container/parts/video-player.test.js [hunk 10]
```diff
@@ -3,6 +3,13 @@ import React from 'react';
 import { mount, shallow } from 'enzyme';
 import VideoPlayer from './video-player';
 import * as storage from '../../../api/local-storage';
+//
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.test.js]: remove comment
